easybuild/framework/easyconfig/tweak.py

```diff
@@ -115,6 +107,16 @@ def tweak(easyconfigs, build_specs, modtool, targetdirs=None):
                 revert_to_regex = True
 
         if not revert_to_regex:
+            # make sure easyconfigs all feature the same toolchain (otherwise we *will* run into trouble)
```

Comment:
can you explain why this only applies when `revert_to_regex = False` ?

Reply:
TBH: I don't know why this restriction really exists.
The only reason I could find is the `source_toolchain` which must (by definition) be a single toolchain. (It is set to the first toolchain found assuming all are the same.) This is only used in this code part, similar `target_toolchain` is also only used here. So after moving those variables down into this codepath (to limit their scope as much as possible) I see no reason to disallow multiple toolchains for the other code paths

On a more general explanation: This code path is used when the toolchain is about to be changed AND map-toolchains is enabled. I can imagine, how this can fail, when multiple toolchains are in effect as the mapping will be much harder.
On the other hand when tweaking ECs via regex-replacement (i.e. NOT this codepath) then all this does not matter: A regex-substitution is a text operation and hence semantic concepts like toolchains do not matter.

FWIW: I tested this on a single install of about 8 different Python from 4 different toolchain generations where I used try-amend to change an EC param and it worked well.

Reply:
I think the reasoning is fine here, it's dangerous to have multiple toolchains in play when we need to deal with a hierarchy, but that won't be the case for a regex tweak